models/get_topics.js

```diff
@@ -8,3 +9,25 @@ exports.selectTopics = () => {
       });
 }
 
+exports.selectArticleById = (article_id) => {
+    
+    if (article_id<=0) {
+        return Promise.reject({ status: 404, msg: "Invalid input, article_id cannot be zero or negative" })
+    }
+    const regex=/\D/
+    const isNonDigit=regex.test(article_id)
+    if(isNonDigit){
+        return Promise.reject({ status: 404, msg: "Invalid input for type integer" })
+    }
+    if (article_id>articleData.length) {
+        return Promise.reject({ status: 404, msg: "No id found, number too high" })
+    }
+    
+    const queryStr = `SELECT * FROM articles WHERE article_id=$1;`
+    return db.query(queryStr,[article_id]).then(({ rows }) => {
+        if(rows.length===0){
+            return Promise.reject({ status: 404, msg: "not found" });
+        }
+        return rows[0];
+      });
```

Comment:
It's best to be a bit more vague with our error messages to prevent giving users too much information about our database. 
We can also use of [PSQL errors](https://notes.northcoders.com/courses/js-back-end/error-handling#4-refactoring-to-individual-error-handling-blocks) which we catch when the user provides the wrong datatype, rather than having to write a custom error handler for each scenario (this will get very repetitive as more endpoints are added).